gjkr/group.go

```diff
@@ -0,0 +1,78 @@
+package gjkr
+
+// group represents the current state of information about the GJKR key
+// generation group. Each GJKR protocol participant should have the same group
+// state at the end of each protocol step.
+type group struct {
+	dishonestThreshold uint16
```

Comment:
Nit: `dishonestThreshold` caused some confusion in `keep-core` in the past. What do you think about leaning on `threshold` and exposing a function `dishonestThreshold` that will compute it as `groupSize - threshold`? This approach feels much more intuitive.